src/NServiceBus.RavenDB/SessionManagement/ContextBagExtensions.cs

```diff
@@ -1,31 +0,0 @@
-﻿namespace NServiceBus.Persistence.RavenDB
-{
-    using System;
-    using NServiceBus.Extensibility;
-    using Raven.Client.Documents.Session;
-
-    static class ContextBagExtensions
-    {
-        /// <summary>
-        /// Retrieves an IAsyncDocumentSession from the ContextBag. If a sessionFunction exists, that will be used
-        /// to create the session. Otherwise, retrieve the session directly from the bag.
-        /// </summary>
-        internal static IAsyncDocumentSession GetAsyncSession(this ContextBag contextBag)
-        {
-            Func<IAsyncDocumentSession> sessionFunction;
-            contextBag.TryGet(out sessionFunction);
```

Comment:
I'm not sure what scenario/configuration could ever hit this code path?

Reply:
It's internal so removing should be fine?

Reply:
yeah but I couldn't find a code path which would ever set such a function?